emscripten floating-point comparison is always false;

compiling to emscripten using cmake: 

```
/gdnative/build/emscripten/_deps/godot-cpp-src/src/core/Rect2.cpp:64:11: error: floating-point comparison is always false; constant cannot be represented exactly in type 'float' [-Werror,-Wliteral-range]
        if (dist == 1e20)
            ~~~~ ^  ~~~~
1 error generated.
```

the error occurs here  :

```
real_t Rect2::distance_to(const Vector2 &p_point) const {
	real_t dist = 1e20;

//......

	if (dist == 1e20)
		return 0;
	else
		return dist;
}
```

am currently doing  `if (dist >= 1e20)` to make it build , but not sure if this is the correct way

--- a/src/core/Rect2.cpp
+++ b/src/core/Rect2.cpp
@@ -46,25 +46,35 @@ namespace godot {
 #endif
 
 real_t Rect2::distance_to(const Vector2 &p_point) const {
-	real_t dist = 1e20;
+	real_t dist = 0.0;
+	bool inside = true;
 
 	if (p_point.x < position.x) {
-		dist = MIN(dist, position.x - p_point.x);
+		real_t d = position.x - p_point.x;
+		dist = d;
+		inside = false;
 	}
 	if (p_point.y < position.y) {
-		dist = MIN(dist, position.y - p_point.y);
+		real_t d = position.y - p_point.y;
+		dist = inside ? d : MIN(dist, d);
+		inside = false;
 	}
 	if (p_point.x >= (position.x + size.x)) {
-		dist = MIN(p_point.x - (position.x + size.x), dist);
+		real_t d = p_point.x - (position.x + size.x);
+		dist = inside ? d : MIN(dist, d);
+		inside = false;
 	}
 	if (p_point.y >= (position.y + size.y)) {
-		dist = MIN(p_point.y - (position.y + size.y), dist);
+		real_t d = p_point.y - (position.y + size.y);
+		dist = inside ? d : MIN(dist, d);
+		inside = false;
 	}
 
-	if (dist == 1e20)
+	if (inside) {
 		return 0;
-	else
+	} else {
 		return dist;
+	}
 }
 
 Rect2 Rect2::clip(const Rect2 &p_rect) const { /// return a clipped rect

